### PR TITLE
Analytics: Updated client creation and edit

### PR DESCRIPTION
**Pull request by OchiengPaul442**
#### Summary of Changes (What does this PR do?)

- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

#### Status of maturity (all need to be checked before merging):

- [ ] I've tested this locally
- [ ] I consider this code done
- [ ] This change ready to hit production in its current state

#### Screenshots (optional)
image
image

## Summary by CodeRabbit

- **New Features**
	- Introduced `DeleteIcon` and `PlusIcon` components for enhanced UI iconography.
	- Enhanced `AddClientForm` to support multiple IP addresses, allowing users to dynamically add or remove fields.
	- Updated `EditClientForm` to manage client data more flexibly with a consolidated `data` prop.
	- Added `displayIPAddresses` function in `UserClientsTable` and `AdminClientsTable` for improved IP address presentation.
	- Improved loading state management for client deactivation actions in `AdminClientsTable`.

- **Bug Fixes**
	- Improved submission logic in forms to filter out empty IP addresses.

- **Style**
	- Minor formatting adjustments for improved readability in various components.

### CODE REVIEW

File: platform/src/common/components/Settings/API/AdminClientsTable.jsx [hunk 1]
```diff
@@ -248,8 +245,7 @@ const AdminClientsTable = () => {
         onClose={() => setConfirmClientDeactivation(false)}
         handleClick={handleDeactivate}
         primaryButtonText={'Deactivate'}
-        loading={isLoadingActivation}
-      >
+        loading={isLoadingActivation}>
```
[the remark this answers is not in the document]
Comment (Baalmart) [reply platform/src/common/components/Settings/API/AdminClientsTable.jsx:248]: Hi @OchiengPaul442 , please review this AI review comment and share feedback accordingly. Thanks!

File: platform/src/common/components/Settings/API/EditClientForm.jsx [hunk 2]
```diff
@@ -105,22 +119,22 @@ const EditClientForm = ({ open, closeModal, cIP = '', cName = '', clientID }) =>
 
       return () => clearTimeout(timer);
     }
-  }, [isError]);
+  }, [isError.isError]);
```
Comment (Baalmart) [inline platform/src/common/components/Settings/API/EditClientForm.jsx:122]: Thanks @OchiengPaul442 :
- Is this variable "isError" always present?
- If not always available, we might want to handle this use case to minimise chances of runtime errors. There are couple of options which must be BACKWARD compatible with earlier version of node (JS).

Thanks!

**Review by Baalmart — COMMENTED** [review]
Thanks @OchiengPaul442 , I have just left a couple of code review comments for your attention. Thanks again, this is great! :)

File: platform/src/common/components/Settings/API/EditClientForm.jsx (hunk 2 above)
Comment (OchiengPaul442) [reply platform/src/common/components/Settings/API/EditClientForm.jsx:122]: Hey @Baalmart ,
The state hook declaration you mentioned is used to manage the error state and display appropriate messages to the user when something goes wrong. This ensures that the error state is always present and updated accordingly whenever an error is detected.

You’re right about the compatibility. This state hook is specifically designed for use in a React frontend and won’t be compatible with a Node.js backend. However, it works perfectly fine for managing error states in a React component.
```
  const [isError, setIsError] = useState({
    isError: false,
    message: '',
    type: '',
  });
```

**Pull request merged**